src/test/java/com/youengineering/openapi/GenerateTest.java

```diff
@@ -69,16 +70,52 @@ private GenerateMojo createGenerateMojo(Framework framework) throws Exception {
     private void assetGeneratedFile_Json(Framework framework) throws Exception {
         JsonNode actualNode = Json.mapper().readTree(getOpenAPIFile(framework, "generated", OutputFormat.json));
         JsonNode expectedNode = Json.mapper().readTree(getOpenAPIFile(framework, "expected", OutputFormat.json));
-        assertEquals(expectedNode, actualNode);
+
+        // Removing all null attributes as openApi deserializer user node.get("extensions") to get value.
+        // And if any of the vales are null, it is getting treated as "null" (String type) and failing with class cast exception
+        removeNullsFromJsonNode(actualNode);
+        removeNullsFromJsonNode(expectedNode);
```

Comment:
I don't remember the details too much honestly, it seems weird that this is necessary? Because then it would have been necessary before, but maybe it changed in the updated swagger-parser-v3 dependency? Still seems weird. But if it only works with this, it should be fine.

Reply:
Earlier here just the JSON nodes were getting asserted.
After the update of dependencies the order of attributes and the paths got changed. The JSON node equals was just telling the objects were not matching and was not able to tell which part was matching. So I decided to parse it into openApi and then assert it. And Somehow it was failing with the above error. (Not sure if it was a problem earlier as well).

Why complicate this? I simply do not want to copy-paste the generated swagger into expected, I just wanted to be very sure the dependency updates do not break anything that is already working fine.